lib/spack/spack/directives.py

```diff
@@ -361,6 +361,8 @@ def _depends_on(pkg, spec, when=None, type=default_deptype, patches=None):
         return
 
     dep_spec = spack.spec.Spec(spec)
+    if not dep_spec.name:
+        raise DependencyError("Invalid dependency specification in package '%s':" % pkg.name, spec)
```

Comment:
OT (not strictly related to the review). When do we need to add a new exception type vs. reuse another one? 

Right now we have a lot of exceptions, which sometimes are used only once or twice. I have always been wondering if we should add a new exception for each new error, or we can use a more generic one (in this case simply `DirectiveError`).

I think a benefit of having just a few is that we might collect them in a common place so people can look at a single file to know which errors can be raised.